src/bodygraph.cpp

```diff
@@ -0,0 +1,577 @@
+        if( width == -1 ) {
+            width = w;
+        } else if( !w_warned && width != w ) {
+            debugmsg( "body_graph \"%s\" defines rows with different widths.", id.c_str() );
+            w_warned = true;
+        } else if( !w_warned && w > BPGRAPH_MAXCOLS ) {
+            debugmsg( "body_graph \"%s\" defines rows with more columns than the maximum (%d).", id.c_str(),
+                      BPGRAPH_MAXCOLS );
+            w_warned = true;
+        }
+        for( int i = ( BPGRAPH_MAXCOLS - w ) / 2; i > 0; i-- ) {
+            r.insert( r.begin(), " " );
+        }
+        for( int i = r.size(); i <= BPGRAPH_MAXCOLS; i++ ) {
+            r.emplace_back( " " );
+        }
+    }
+
+    for( int i = ( BPGRAPH_MAXROWS - rows.size() ) / 2; i > 0; i-- ) {
+        std::vector<std::string> r;
+        for( int j = 0; j < BPGRAPH_MAXCOLS; j++ ) {
+            r.emplace_back( " " );
+        }
+        rows.insert( rows.begin(), r );
+    }
+
+    for( int i = rows.size(); i <= BPGRAPH_MAXROWS; i++ ) {
+        std::vector<std::string> r;
+        for( int j = 0; j < BPGRAPH_MAXCOLS; j++ ) {
+            r.emplace_back( " " );
+        }
+        rows.emplace_back( r );
+    }
+}
+
+void bodygraph::check() const
+{
+    if( parts.empty() ) {
+        debugmsg( "body_graph \"%s\" defined without parts.", id.c_str() );
+    }
+    for( const auto &bgp : parts ) {
+        if( utf8_width( bgp.first ) > 1 ) {
+            debugmsg( "part \"%s\" in body_graph \"%s\" is more than 1 character.", bgp.first, id.c_str() );
+        }
+        if( bgp.second.bodyparts.empty() && bgp.second.sub_bodyparts.empty() ) {
+            debugmsg( "part \"%s\" in body_graph \"%s\" contains no body_parts or sub_body_parts definitions.",
+                      bgp.first, id.c_str() );
+        }
+    }
+}
+
+/***************************** Display routines *****************************/
+
+/* Basic layout. Not accurate or final, but illustrates the basic idea.
+
+     parts list                      body graph                  part info
+<- variable width -><------------ fixed width (40) --------><- variable width ->
+
++------------------+----------------Body Status-------------+------------------+
+| >Head            |                                        | Worn:            |
+|  Torso           |                     O                  |  Baseball cap,   |
+|  L. Arm          |                   #####                |  Eyeglasses      |
+|  R. Arm          |                  #######               | Coverage: ~60%   |
+|  L. Hand         |                 ## ### ##              | Encumbrance: 8   |
+|  R. Hand         |                 #  ###  #              | Protection:      |
+...
+
+*/
+
+#define BPGRAPH_HEIGHT 24
+
+bodygraph_info::bodygraph_info()
+{
+    std::vector<std::string> worn_names = {};
+    avg_coverage = 0;
+    total_encumbrance = 0;
+    worst_case = resistances();
+    median_case = resistances();
+    best_case = resistances();
+}
```

Comment:
`avg_coverage` and `total_encumbrance` can be initialized in the struct declaration and the rest of the block is redundant
```suggestion
```